Fix try builds of the system add-on

Osmose opened the pull request:
There are failures with our add-on that can happen only during
packaging of Firefox, such as duplicate bootstrap.js files. We should
run that step in TaskCluster CI so we can catch those issues early.

This should catch the issue from #718 and fail. I'm still working on a fix, but want to run this in the meantime.

mythmon commented on recipe-client-addon/bin/tc/package.sh:
```diff
@@ -0,0 +1,19 @@
+source /root/.cargo/env
+
+echo 'Packaging Firefox'
+./mach package
```
Might as well put the results of this in an artifact?

Osmose replied:
Sure, if we can find a good way to determine the name of that directory. I wouldn't consider it a blocker though.

Osmose commented:
Okay, this is ready. The diff looks big, but most of the changes are just copy-pastes from bootstrap.js to ShieldRecipeClient.jsm.

gijsk commented on recipe-client-addon/bootstrap.js:
```diff
@@ -4,95 +4,27 @@
 
+  // Unload add-on modules. We don't do this in ShieldRecipeClient so that
+  // modules are not unloaded accidentally during tests.
+  const log = LogManager.getLogger("bootstrap");
```
We may or may not have called LogManager.configure() here, per https://bugzilla.mozilla.org/show_bug.cgi?id=1351465#c5 . I don't know if that should be a serious concern, but I wanted to flag this up.

Osmose commented:
Updated. I moved the logging setup to before the early return. There's some opportunity for refactoring this to be better but I'm punting on that for now.
